inc/classes/class-srm-post-type.php

```diff
@@ -448,9 +462,50 @@ public function action_save_post( $post_id ) {
 			 * redirect info is saved, updating the cache before it is not sufficient.
 			 */
 			srm_flush_cache();
+
+			add_filter( 'redirect_post_location', array( $this, 'add_notice_query_var' ), 99, 2 );
 		}
 	}
 
+	/**
+	 * Add custom query var to indicate that this is a duplicate redirection rule.
+	 *
+	 * @param string $location The destination URL.
+	 * @param int    $post_id  The post ID.
+	 *
+	 * @return string
+	 */
+	public function add_notice_query_var( $location, $post_id ) {
+
+		remove_filter( 'redirect_post_location', array( $this, 'add_notice_query_var' ), 99 );
+
+		global $wpdb;
+
+		$post_type = get_post_type( $post_id );
+
+		if ( 'redirect_rule' !== $post_type ) {
+			return $location;
+		}
+
+		$redirect_from = get_post_meta( $post_id, '_redirect_rule_from', true );
+
+		if ( empty( $redirect_from ) ) {
+			return $location;
+		}
+
+		// Get post if exist with same redirection rule.
+		$existing_post_id = $wpdb->get_var( $wpdb->prepare( "SELECT post_id FROM $wpdb->postmeta WHERE meta_key=%s AND meta_value=%s", '_redirect_rule_from', $redirect_from ) );
```

Comment:
can we use a WP_Query here instead of a direct $wp_db query? we can limit to fields=['id']